### ISSUE TITLE
[java]UseCollectionIsEmpty can not detect the case foo.bar().size()

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2542]
**Affects PMD Version:** 
6.22.0
**Rule:**
UseCollectionIsEmpty

**Description:**
UseCollectionIsEmpty can not detect the case <code>foo.bar().size()</code>.
AST analyse result of the code sample:
```
<PrimaryPrefix FindBoundary='false' Image='' SingleLine='true' SuperModifier='false' ThisModifier='false'>
	<Name FindBoundary='false' Image='vr.getMembers' SingleLine='true' />
</PrimaryPrefix>
<PrimarySuffix ArgumentCount='0' Arguments='true' ArrayDereference='false' FindBoundary='false' Image='' SingleLine='true'>
    <Arguments ArgumentCount='0' FindBoundary='false' Image='' SingleLine='true' Size='0' />
</PrimarySuffix>
    <PrimarySuffix ArgumentCount='-1' Arguments='false' ArrayDereference='false' FindBoundary='false' Image='size' SingleLine='true' />
<PrimarySuffix ArgumentCount='0' Arguments='true' ArrayDereference='false' FindBoundary='false' Image='' SingleLine='true'>
    <Arguments ArgumentCount='0' FindBoundary='false' Image='' SingleLine='true' Size='0' />
</PrimarySuffix>
```
In the case <code>vr.getMembers().size()</code>, PrimaryPrefix is <code>vr.getMembers</code>, may cause error in comparing whether PrimaryPrefix is collection

**Code Sample demonstrating the issue:**

```
while (vr.getMembers().size() > 0)
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.
